File: source/backend/python/application/domain/service/send_money_service.py
from application.port.ingress.send_money_command import SendMoneyCommand
from application.port.ingress.send_money_use_case import SendMoneyUseCase

class SendMoneyService(SendMoneyUseCase):
    def send_money(self, send_money_command: SendMoneyCommand) -> bool:
        # モデルの更新
        send_money_command.source_account.withdraw(send_money_command.money)
        send_money_command.target_account.deposit(send_money_command.money)

        # DBへの保存

        return True

File: source/backend/typescript/application/domain/service/SendMoneyService.ts
/**
 * 送金処理の具象クラス(処理の実態)
 */

import { SendMoneyUseCase } from "../../port/in/SendMoneyUseCase";
import { SendMoneyCommand } from "../../port/in/SendMoneyCommand";

export class SendMoneyService implements SendMoneyUseCase {
    sendMoney(sendMoneyCommand: SendMoneyCommand) {
        // 引数はSendMoneyCommandクラスでヴァリデーションがされている(入力値の妥当性確認済)
        
        // モデルの更新
        sendMoneyCommand.sourceAccount.withdraw(sendMoneyCommand.sourceAccount, sendMoneyCommand.money)
        sendMoneyCommand.targetAccount.deposit(sendMoneyCommand.sourceAccount, sendMoneyCommand.money)

        // DBへの保存
        
        return true;
    }
}